From 33b1b55ed52c1f7d6dac1b382058d82188ef524c Mon Sep 17 00:00:00 2001
From: anonym <anonym@riseup.net>
Date: Wed, 27 Feb 2019 09:30:10 +0100
Subject: [PATCH] Invalidate config when restarting autoconfiguration.

Otherwise residual data from previous attempts can leak into following
attempts, e.g.:

1. Try "a@a.a"
2. All methods fail, so we end up in "manual edit" mode
3. Note that Server hostname == "a.a", which is expected
4. Then start over by changing the email to "b@b.b"
5. All methods fail, so we end up in "manual edit" mode
6. Note that Server hostname == "a.a", but should be "b.b"
---
 comm/mail/components/accountcreation/content/emailWizard.js | 1 +
 1 file changed, 1 insertion(+)

--- a/comm/mail/components/accountcreation/content/emailWizard.js
+++ b/comm/mail/components/accountcreation/content/emailWizard.js
@@ -406,6 +406,7 @@ EmailConfigWizard.prototype =
    */
   onStartOver : function()
   {
+    this._currentConfig = null;
     if (this._abortable) {
       this.onStop();
     }
